<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary
import pandas as pd

data={
    'Nama':['Nur', 'Azizah', 'Aisah', 'Munir'],
    'Umur':[20, 56, 30, 23],
    'Kota':['Tangerang', 'Pasuruan', 'Bogor', 'Depok']
}

df=pd.DataFrame(data)
print(df)

     Nama  Umur       Kota
0     Nur    20  Tangerang
1  Azizah    56   Pasuruan
2   Aisah    30      Bogor
3   Munir    23      Depok


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [4]:
# Membaca data dan memasukannya kedalam suatu variabel
df=pd.read_csv('diamonds.csv')
df

Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:



*   Sumber data: Series dengan list dibuat dengan data yang terurut dalam list, dan index ditentukan secara eksplisit oleh pengguna. Series dengan dictionary dibuat dengan data dalam bentuk pasangan key-value, di mana key menjadi index dan value menjadi data dalam Series. Index: Pada Series dengan list, index ditentukan secara manual oleh pengguna sesuai kebutuhan. Pada Series dengan dictionary, index secara otomatis diambil dari key dictionary, yang lebih praktis ketika data sudah berbentuk key value.
*   Series adalah struktur data satu dimensi. Mirip seperti satu kolom data dalam sebuah tabel, dengan setiap elemen memiliki index bawaan. Sedangkan DataFrame adalah struktur data dua dimensi yang bisa dianggap seperti tabel Excel. DataFrame memiliki baris dan kolom, yang memungkinkan Anda untuk menyimpan dan bekerja dengan data yang lebih kompleks.



In [6]:
# code here
import pandas as pd

data_list = [10, 20, 30, 40]
series_list = pd.Series(data_list, index=['A', 'B', 'C', 'D'])
print("Series dari List:")
print(series_list)

data_dict = {'A': 100, 'B': 200, 'C': 300}
series_dict = pd.Series(data_dict)
print("\nSeries dari Dictionary:")
print(series_dict)

Series dari List:
A    10
B    20
C    30
D    40
dtype: int64

Series dari Dictionary:
A    100
B    200
C    300
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [10]:
# code here
df=pd.read_csv('diamonds.csv')

print("Data Pertama:")
print(df.head())

print("\nData Terakhir:")
print(df.tail())

first_price=df['price'].iloc[0]
print("\nBaris pertama kolom price:", first_price)

last_color=df['color'].iloc[0]
print("\nBaris terakhir kolom color:", last_color)

Data Pertama:
   Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75  

Data Terakhir:
       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75     

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab:

 Deskriptif Statistik adalah metode yang digunakan untuk menggambarkan data yang telah terkumpul, tanpa bermaksud membuat kesimpulan yang berlaku secara umum. Statistik deskriptif dapat digunakan untuk menyajikan data dalam bentuk tabel, diagram, grafik, atau besaran lainnya. Dapat dipecah menjadi ukuran kecenderungan sentral dan ukuran variabilitas. Ukuran kecenderungan sentral meliputi mean, median, dan modus. Sedangkan ukuran variabilitas meliputi deviasi standar, varians, variabel minimum dan maksimum, kurtosis, dan kemiringan.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [12]:
# code here
import pandas as pd

df = pd.read_csv('diamonds.csv')
desc_stats = df.describe()
print("Deskriptif Statistik:\n", desc_stats)

mean_price = df['price'].mean()
print("\nRata-rata dari kolom price:", mean_price)

std_depth = df['depth'].std()
print("\nStandar deviasi dari kolom depth:", std_depth)

max_carat = df['carat'].max()
print("\nNilai maksimum dari kolom carat:", max_carat)


Deskriptif Statistik:
          Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.71

Jawab:

Rata-rata dari kolom price: 3932.799721913237

Standar deviasi dari kolom depth: 1.4326213188337733

Nilai maksimum dari kolom carat: 5.01